### PR TITLE
Fix sensitive

### PR DESCRIPTION
**Pull request by axl89**
Fixes `wazuh agent auth` resource leaking sensitive data and removes duplicated code:

```
  * execute[/var/ossec/bin/agent-auth -m 1.2.3.4 -p 1515 -A hostname -P xxxxxxxxx -P xxxxxxxxx] action run[2020-12-18T13:08:31+00:00] INFO: Processing execute[/var/ossec/bin/agent-auth -m 1.2.3.4 -p 1515 -A hostname -P xxxxxxxxx -P xxxxxxxxx] action run (wazuh_agent::agent line 92)

    [execute] 2020/12/18 13:08:31 agent-auth: INFO: Started (pid: 15855).
              2020/12/18 13:08:31 agent-auth: INFO: Connected to 1.2.3.4:1515
              2020/12/18 13:08:31 agent-auth: INFO: Using agent name as: hostname
              2020/12/18 13:08:31 agent-auth: INFO: Send request to manager. Waiting for reply.
              2020/12/18 13:08:31 agent-auth: INFO: Received response with agent key
              2020/12/18 13:08:31 agent-auth: INFO: Valid key created. Finished.
              2020/12/18 13:08:31 agent-auth: INFO: Connection closed.
[2020-12-18T13:08:31+00:00] INFO: execute[/var/ossec/bin/agent-auth -m 1.2.3.4 -p 1515 -A hostname -P xxxxxxxxx -P xxxxxxxxx] ran successfully
    - execute /var/ossec/bin/agent-auth -m 1.2.3.4 -p 1515 -A hostname -P xxxxxxxxx -P xxxxxxxxx
```

### CODE REVIEW

File: cookbooks/wazuh_agent/recipes/agent.rb [hunk 1]
```diff
@@ -85,14 +85,12 @@
   args << ' -k ' + agent_auth['key']
 end
 
-if agent_auth['password']
```
Comment (axl89) [inline cookbooks/wazuh_agent/recipes/agent.rb:88]: This piece of code is duplicated on [L71](https://github.com/wazuh/wazuh-chef/pull/119/files#diff-fffce9a3886a2b4d0a739df924002b58e24e23fffb62c21e0f54ab2aec0b6fc2L71).

**Comment by sergiogp98** [comment]
Thanks @axl89 ! We will merge this in soon.

**Comment by axl89** [comment]
:cry:

**Comment by Zenidd** [comment]
Hey @axl89, 

Sorry for the slowness. I just reopened the PR(https://github.com/wazuh/wazuh-chef/pull/126) using an internal branch so the Github Actions tests get launched against your commits. Also, I used `master` (our develop branch) as the base branch. Thank you for your contribution mate!

_-JP_

**Pull request closed without merging**